### PR TITLE
Add Chroma to Vector Database examples

### PR DESCRIPTION
**Pull request by atroyn**
This PR adds Chroma to the vector database examples in the OpenAI cookbook.

[Chroma](docs.trychroma.com) is a lightweight, developer-friendly embeddings store which aims to get A.I application developers up and running with embeddings as fast as possible. 

This PR:

- Adds Chroma as another example to  the generic `Using_vector_databases_for_embeddings_search.ipynb`
- Adds an example which shows some of the different tradeoffs in using document retrieval with an embeddings store for LLM-based document Q/A. This uses the `gpt3.5-turbo` api as well as the `ada-002` embeddings API. 
- Adds a scientific question/answering dataset for use with the above example, in `jsonl` format.

### CODE REVIEW

**Comment by ted-at-openai** [comment]
Thanks for the PR; I'll take a look. Always happy to include more solutions for end users.

File: examples/vector_databases/chroma/hyde-with-chroma-and-openai.ipynb [hunk 1]
```diff
@@ -0,0 +1,984 @@
...
+    "corpus_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading the corpus into Chroma\n",
+    "\n",
+    "The next step is to load the Corpus into Chroma. Given an embedding function, Chroma will automatically handle embedding each document, and will store it alongside its text and metadata, making it simple to query. "
```
Comment (csvoss) [inline examples/vector_databases/chroma/hyde-with-chroma-and-openai.ipynb:523]:
```suggestion
    "The next step is to load the corpus into Chroma. Given an embedding function, Chroma will automatically handle embedding each document, and will store it alongside its text and metadata, making it simple to query. "
```

File: examples/vector_databases/chroma/hyde-with-chroma-and-openai.ipynb [hunk 2]
```diff
@@ -0,0 +1,984 @@
...
+    "gpt_inferred = assess_claims(claims)\n",
+    "confusion_matrix(gpt_inferred, groundtruth)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Results\n",
+    "From these results we see that the LLM is strongly biased to asess claims as true, even when they are false, and also tends to assess false claims as not having enough evidence. Note that 'not enough evidence' is with respect to the model's assessment of the claim in a vacuum, without additional context."
```
Comment (csvoss) [inline examples/vector_databases/chroma/hyde-with-chroma-and-openai.ipynb:399]:
```suggestion
    "From these results we see that the LLM is strongly biased to assess claims as true, even when they are false, and also tends to assess false claims as not having enough evidence. Note that 'not enough evidence' is with respect to the model's assessment of the claim in a vacuum, without additional context."
```

**Review by csvoss — COMMENTED** [review]
This was a fun read! I feel like I actually understand how the hallucinated abstracts improve the overall performance, thanks to the notes and commentary. 

This code is outside my regular scope but I happened to be here looking this notebook over anyways, so I left notes on a few minor spelling nits that I noticed while reading.

**Comment by atroyn** [comment]
Thanks @csvoss for catching those spelling nits!

**Comment by atroyn** [comment]
@ted-at-openai - let me know if there are any changes you'd like me to make here!

**Comment by ted-at-openai** [comment]
I'll take a look - thanks for the ping.

**Comment by atroyn** [comment]
@ted-at-openai following up here again - i'm concerned that this will grow stale soon, we'd like to land this ahead of the retrieval plugin integration.

**Comment by ted-at-openai** [comment]
Thanks for following up. Here is my current thinking:

Right now the PR can't be merged because due to merge conflicts stemming from another update made to the notebook.

Ideally I'd like to have one notebook per provider, so we don't have a huge blob of concatenated examples, which is (a) hard to read and (b) hard to have multiple people merge to.

Option 1:

- You can fix the merge conflicts by applying your changes to the updated notebook (sorry for the extra work)

Option 2:

- You can wait for us to disentangle the notebook and try to apply your commits to that new framework (this will take longer).

Option 1 will go fastest, but take more work from your side.

I'm sorry I haven't had the time to do this; unfortunately it's not my at the top of my priority list right now. I still really appreciate your contributions, am happy to merge them as soon as ready, and am happy to continue discussing.

**Comment by atroyn** [comment]
Happy to resolve the conflict on our end

**Comment by atroyn** [comment]
All set.

File: examples/vector_databases/Using_vector_databases_for_embeddings_search.ipynb [hunk 3]
```diff
@@ -101,16 +104,16 @@
     "# Redis client library for Python\n",
     "import redis\n",
     "\n",
-    "# Pinecone's client library for Python\n",
+    "# Chroma's client library for Python\n",
```
Comment (ted-at-openai) [inline examples/vector_databases/Using_vector_databases_for_embeddings_search.ipynb:107]: Here it looks like Chroma is just single commented, but the others are double commented - should they all be single commented?

Comment (atroyn) [reply examples/vector_databases/Using_vector_databases_for_embeddings_search.ipynb:107]: oops, fixed.

**Review by ted-at-openai — APPROVED** [review]

**Comment by ted-at-openai** [comment]
Merged! Thanks for your patience during the long delay. Appreciate your quick changes. Glad to you have your example in the repo.

**Comment by atroyn** [comment]
Thanks for reviewing! I know it's a busy time, much appreciated.

**Review by Honey980 — APPROVED** [review]

**Pull request merged**